apps/storybook-react/README.md

```diff
@@ -1,17 +1,24 @@
-# `@metamask/storybook-react`
```

Comment:
Updating storybook readme to serve as a better description of what this package is for and the introduction page for storybook

<img width="1512" alt="Screenshot 2024-11-27 at 10 47 39 AM" src="https://github.com/user-attachments/assets/4a75c0cb-13d0-4ee6-b050-2471198e2739">